### DISCUSSION TITLE
Feature Request: Implement RAND(seed) function

### DISCUSSION THREAD

**Discussion by waltersf** [duckdb/duckdb#5843] [Ideas]
Why do you want this feature?

MySQL, SQL Server and others implement a function `RAND(seed)`. DuckDB has the function `RANDOM()` and to initialized the random number generator, we must use `SETSEED(seed)`. A single function `RAND(seed)` call would avoid creating unnecessary code when executing a SELECT command. For example:
```sql
SELECT *, RAND(131) FROM table
-- instead of:
SELECT *, SETSEED(131), RANDOM() FROM table
```

Also, I could not find a way to use (correctly) `SETSEED(seed)` and `RANDOM()` in same command, as the second SELECT above. For instance:

```py
>>> conn.query('select setseed(.1), random()').df()
   setseed(.1)  random()
0          NaN  0.896351
>>> conn.query('select setseed(.2), random()').df() # Returns same results as before (incorrect)
   setseed(.2)  random()
0          NaN  0.896351
>>> conn.query('select setseed(.2), random()').df() # Returns correct result
   setseed(.2)  random()
0          NaN  0.599364

```
Only after the second execution setting seed to .2, the `RANDOM()` function generated a new number. 

According to this post https://stackoverflow.com/questions/24256258/order-by-random-with-seed-in-sqlite, SQLite suffers from the same problem (my use case is the same, I know that it is inefficient). 

I found an example using PostgreSQL and `SETSEED` function that is working: https://database.guide/how-setseed-works-in-postgresql/ . 
Please, review if this issue should be classified as a new feature (new `RAND(seed)` function) or a bug in `SETSEED()`.

TIA

**Comment by Mytherin** [comment]
Happy to review a PR that adds an optional `seed` argument to the `random` function.

The seed for `random` is determined before execution of the actual function - as such using both `setseed` and `random` does indeed not work in your example. `setseed` has to be used in a separate SQL statement instead.
